### PR TITLE
Use dynamic versions in canary upgrade docs

### PR DESCRIPTION
**Pull request by SRodi**
* remove hardcoded revision references
* use dynamically created revisions
* update test scripts accordingly

- [ ] Ambient
- [x] Docs
- [x] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Extensions and Telemetry
- [ ] Security
- [x] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

### CODE REVIEW

**Comment by SRodi** [comment]
@frankbu I have requested edit access to the following files so that I can update diagrams to use generic references `OLD_REVISION` and `NEW_REVISION` instead of static `1-9-5` and `1-10-0`.  This way diagrams would be generic so that we don't have to update them in future.

* https://docs.google.com/drawings/d/1eNSg78LzpLaK2p0v1rczFxaFVYWUEustqNxVSS97PjE/edit
* https://docs.google.com/drawings/d/16FTMWw_fvG1w1Mk4HnytZ-qY0o4enflhtzIXkxrv_i4/edit

File: content/en/boilerplates/revision-tags-middle.md [hunk 1]
```diff
@@ -12,11 +12,11 @@ The cluster operator can view this mapping in addition to tagged namespaces thro
 {{< text bash >}}
 $ istioctl tag list
 TAG         REVISION NAMESPACES
-default     1-9-5    ...
-prod-canary 1-10-0   ...
-prod-stable 1-9-5    ...
+default     {{< istio_previous_version_revision >}}-5   ...
```
Comment (frankbu) [inline content/en/boilerplates/revision-tags-middle.md:15]: There often is no `1-x-5` version of a previous release when a new one comes out. The only guarantee is that there is a `1-x-0`, although it's probably safe to assume there was at least one patch release, so I would use 0 or 1 for the previous revision:

```suggestion
default     {{< istio_previous_version_revision >}}-1   ...
```

File: content/en/docs/setup/upgrade/canary/index.md [hunk 2]
```diff
@@ -41,7 +41,7 @@ To install a new revision called `canary`, you would set the `revision` field as
 
 {{< tip >}}
 In a production environment, a better revision name would correspond to the Istio version.
-However, you must replace `.` characters in the revision name, for example, `revision=1-6-8` for Istio `1.6.8`,
+However, you must replace `.` characters in the revision name, for example, `revision=1-17-2` for Istio `1.17.2`,
```
[the remark this answers is not in the document]
Comment (SRodi) [reply content/en/docs/setup/upgrade/canary/index.md]: addressed in 5627df6518b1b7a290183e7b59ff1388afaa875e

File: content/en/boilerplates/revision-tags-usage.md [hunk 3]
```diff
@@ -1,5 +1,5 @@
 ---
 ---
-Consider a cluster with two revisions installed, `1-9-5` and `1-10-0`. The cluster operator creates a revision tag `prod-stable`,
-pointed at the older, stable `1-9-5` version, and a revision tag `prod-canary` pointed at the newer `1-10-0` revision. That
+Consider a cluster with two revisions installed, `{{< istio_previous_version_revision >}}-5` and `{{< istio_full_version_revision >}}`. The cluster operator creates a revision tag `prod-stable`,
+pointed at the older, stable `{{< istio_previous_version_revision >}}-5` version, and a revision tag `prod-canary` pointed at the newer `{{< istio_full_version_revision >}}` revision. That
```
Comment (frankbu) [inline content/en/boilerplates/revision-tags-usage.md:4]:
```suggestion
Consider a cluster with two revisions installed, `{{< istio_previous_version_revision >}}-1` and `{{< istio_full_version_revision >}}`. The cluster operator creates a revision tag `prod-stable`,
pointed at the older, stable `{{< istio_previous_version_revision >}}-1` version, and a revision tag `prod-canary` pointed at the newer `{{< istio_full_version_revision >}}` revision. That
```

File: content/en/docs/setup/upgrade/canary/index.md [hunk 4]
```diff
@@ -159,15 +159,15 @@ If you're using Helm, refer to the [Helm upgrade documentation](/docs/setup/upgr
...
     NAME                                CLUSTER        CDS        LDS        EDS        RDS        ECDS         ISTIOD                             VERSION
-    sleep-78ff5975c6-62pzf.app-ns-3     Kubernetes     SYNCED     SYNCED     SYNCED     SYNCED     NOT SENT     istiod-1-10-0-7f6fc6cfd6-s8zfg     1.16.1
-    sleep-78ff5975c6-8kxpl.app-ns-1     Kubernetes     SYNCED     SYNCED     SYNCED     SYNCED     NOT SENT     istiod-1-9-5-bdf5948d5-n72r2       1.16.1
-    sleep-78ff5975c6-8q7m6.app-ns-2     Kubernetes     SYNCED     SYNCED     SYNCED     SYNCED     NOT SENT     istiod-1-9-5-bdf5948d5-n72r2       1.16.1
+    sleep-78ff5975c6-62pzf.app-ns-3     Kubernetes     SYNCED     SYNCED     SYNCED     SYNCED     NOT SENT     istiod-{{< istio_full_version_revision >}}-7f6fc6cfd6-s8zfg     1.16.1
+    sleep-78ff5975c6-8kxpl.app-ns-1     Kubernetes     SYNCED     SYNCED     SYNCED     SYNCED     NOT SENT     istiod-{{< istio_previous_version_revision >}}-1-bdf5948d5-n72r2      1.16.1
+    sleep-78ff5975c6-8q7m6.app-ns-2     Kubernetes     SYNCED     SYNCED     SYNCED     SYNCED     NOT SENT     istiod-{{< istio_previous_version_revision >}}-1-bdf5948d5-n72r2      1.16.1
```
Comment (frankbu) [inline content/en/docs/setup/upgrade/canary/index.md:164]:
```suggestion
    sleep-78ff5975c6-62pzf.app-ns-3     Kubernetes     SYNCED     SYNCED     SYNCED     SYNCED     NOT SENT     istiod-{{< istio_full_version_revision >}}-7f6fc6cfd6-s8zfg     {{< istio_full_version_revision >}}
    sleep-78ff5975c6-8kxpl.app-ns-1     Kubernetes     SYNCED     SYNCED     SYNCED     SYNCED     NOT SENT     istiod-{{< istio_previous_version_revision >}}-1-bdf5948d5-n72r2      {{< istio_previous_version_revision >}}.1
    sleep-78ff5975c6-8q7m6.app-ns-2     Kubernetes     SYNCED     SYNCED     SYNCED     SYNCED     NOT SENT     istiod-{{< istio_previous_version_revision >}}-1-bdf5948d5-n72r2      {{< istio_previous_version_revision >}}.1
```

File: content/en/docs/setup/upgrade/canary/index.md [hunk 5]
```diff
@@ -182,27 +182,27 @@ Verify the application to control plane mapping using `istioctl proxy-status` co
...
-sleep-5984f48bc7-kmj6x.app-ns-1                        Kubernetes     SYNCED     SYNCED     SYNCED     SYNCED       NOT SENT     istiod-1-10-0-7f6fc6cfd6-jsktb     1.16.1
-sleep-78ff5975c6-jldk4.app-ns-3                        Kubernetes     SYNCED     SYNCED     SYNCED     SYNCED       NOT SENT     istiod-1-10-0-7f6fc6cfd6-jsktb     1.16.1
-sleep-7cdd8dccb9-5bq5n.app-ns-2                        Kubernetes     SYNCED     SYNCED     SYNCED     SYNCED       NOT SENT     istiod-1-10-0-7f6fc6cfd6-jsktb     1.16.1
+sleep-5984f48bc7-kmj6x.app-ns-1                        Kubernetes     SYNCED     SYNCED     SYNCED     SYNCED       NOT SENT     istiod-{{< istio_full_version_revision >}}-7f6fc6cfd6-jsktb     1.16.1
+sleep-78ff5975c6-jldk4.app-ns-3                        Kubernetes     SYNCED     SYNCED     SYNCED     SYNCED       NOT SENT     istiod-{{< istio_full_version_revision >}}-7f6fc6cfd6-jsktb     1.16.1
+sleep-7cdd8dccb9-5bq5n.app-ns-2                        Kubernetes     SYNCED     SYNCED     SYNCED     SYNCED       NOT SENT     istiod-{{< istio_full_version_revision >}}-7f6fc6cfd6-jsktb     1.16.1
```
Comment (frankbu) [inline content/en/docs/setup/upgrade/canary/index.md:187]:
```suggestion
sleep-5984f48bc7-kmj6x.app-ns-1                        Kubernetes     SYNCED     SYNCED     SYNCED     SYNCED       NOT SENT     istiod-{{< istio_full_version_revision >}}-7f6fc6cfd6-jsktb     {{< istio_full_version >}}
sleep-78ff5975c6-jldk4.app-ns-3                        Kubernetes     SYNCED     SYNCED     SYNCED     SYNCED       NOT SENT     istiod-{{< istio_full_version_revision >}}-7f6fc6cfd6-jsktb     {{< istio_full_version >}}
sleep-7cdd8dccb9-5bq5n.app-ns-2                        Kubernetes     SYNCED     SYNCED     SYNCED     SYNCED       NOT SENT     istiod-{{< istio_full_version_revision >}}-7f6fc6cfd6-jsktb     {{< istio_full_version >}}
```

File: content/en/boilerplates/revision-tags-usage.md (hunk 3 above)
Comment (SRodi) [reply content/en/boilerplates/revision-tags-usage.md]: addressed in ed722bf8e9211aa0ec9da36163ae8e3bd11ffb6d

File: content/en/docs/setup/upgrade/canary/index.md (hunk 5 above)
Comment (SRodi) [reply content/en/docs/setup/upgrade/canary/index.md]: addressed in ed722bf8e9211aa0ec9da36163ae8e3bd11ffb6d

File: content/en/docs/setup/upgrade/helm/canaryupgrade_test.sh [hunk 6]
```diff
@@ -13,13 +14,17 @@
...
 source "content/en/docs/setup/upgrade/helm/common.sh"
+source "content/en/boilerplates/snips/args.sh"
 
 set -e
 set -u
 
 set -o pipefail
 
-# @setup profile=none
+# @setup profile=non
```
Comment (frankbu) [inline content/en/docs/setup/upgrade/helm/canaryupgrade_test.sh:24]:
```suggestion
# @setup profile=none
```

Comment (SRodi) [reply content/en/docs/setup/upgrade/helm/canaryupgrade_test.sh]: addressed in 52f8300bf1c5262d40d69f12b36317cd5b4c9fee

**Review by frankbu — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**